### DISCUSSION TITLE
Why doesn't the "Send files using a metadata field" feature work

### DISCUSSION THREAD

**Discussion by Artlesbol** [ObsidianPublisher/obsidian-github-publisher#258] [Q&A — Plugin] [answered]
I try to use the "Send files using a metadata field" feature in "Attachment & embed note config" to upload my images. I set the metadata field "banner" in frontmatter and set it in plugin settings, like in the picture below.
![image](https://github.com/ObsidianPublisher/obsidian-github-publisher/assets/74887228/d0608091-fc2b-48d0-811c-dfd92f1c0be6)
then I try to publish this test document, which includes a banner image, but the image isn't uploaded to my repository.
![image](https://github.com/ObsidianPublisher/obsidian-github-publisher/assets/74887228/826b6dbd-89aa-448a-b8b3-657e1690bbc6)
![image](https://github.com/ObsidianPublisher/obsidian-github-publisher/assets/74887228/fb5f0012-999e-4e94-bfda-e482b88440de)
![image](https://github.com/ObsidianPublisher/obsidian-github-publisher/assets/74887228/146967b5-b580-46fc-8dfc-86636876cecb)
I'd appreciate it if anyone could tell me how to make it work.

**Comment by Mara-Li** [comment, marked as answer]
You need to use a valid path.
The path you use is not valid in Obsidian.

The best way is using a obsidian links like `[[my file]]`

**Reply by Artlesbol** [reply]
thank you for your timely anwser.
but it still not work after i change the path to a obsidian links.
I have tried these format of links. All of them just push md file itself without image file. Are there any other valid path can i try? 
![image](https://github.com/ObsidianPublisher/obsidian-github-publisher/assets/74887228/2eb8e560-98cd-4512-80d9-b8c6de699fee)
![image](https://github.com/ObsidianPublisher/obsidian-github-publisher/assets/74887228/0f963976-7055-4d13-8fc3-2db840c8e056)
![image](https://github.com/ObsidianPublisher/obsidian-github-publisher/assets/74887228/9719291b-f571-4b77-a721-aced279885ac)
![image](https://github.com/ObsidianPublisher/obsidian-github-publisher/assets/74887228/2d123ce7-e014-4348-b6c9-bd761c4f832a)

**Comment by Mara-Li** [comment]
🤔 
Could you please open the logs ? Maybe there is an unknown error I need to fix :D

**Reply by Artlesbol** [reply]
of course. I capture this log for you to locate the possible bug.
if you need more information, please tell me.
![image](https://github.com/ObsidianPublisher/obsidian-github-publisher/assets/74887228/e94254ae-6713-49a4-a7b2-2025de3bb865)
```text
plugin:obsidian-mkdocs-publisher:21 [GitHub Publisher](pr.eval):
 Repository Artlesbol/blog.artlesbol.github.io exists. Now testing the main branch.
plugin:obsidian-mkdocs-publisher:21 [GitHub Publisher](pr.eval):
 Branch successfully created (status: 201) on Artlesbol/blog.artlesbol.github.io
plugin:obsidian-mkdocs-publisher:21 [GitHub Publisher](pr.eval):
 Upload 测试文件3.md:content/post/测试文件3.md on Artlesbol/blog.artlesbol.github.io:obsidian-blog-artlesbol-top-12-22-2023
api.github.com/repos/Artlesbol/blog.artlesbol.github.io/actions/workflows/hugo.yml/dispatches:1     Failed to load resource: the server responded with a status of 404 ()
```
and here is my config json
```json
{
  "github": {
    "branch": "main",
    "automaticallyMergePR": true,
    "tokenPath": "%configDir%/plugins/%pluginID%/env",
    "api": {
      "tiersForApi": "Github Free/Pro/Team (default)",
      "hostname": ""
    },
    "workflow": {
      "commitMessage": "[PUBLISHER] Merge",
      "name": "hugo.yml"
    },
    "verifiedRepo": true
  },
  "upload": {
    "behavior": "obsidian",
    "defaultName": "content/post",
    "rootFolder": "",
    "yamlFolderKey": "",
    "frontmatterTitle": {
      "enable": false,
      "key": "title"
    },
    "replaceTitle": [],
    "replacePath": [],
    "autoclean": {
      "enable": false,
      "excluded": []
    },
    "folderNote": {
      "enable": false,
      "rename": "index.md",
      "addTitle": {
        "enable": false,
        "key": "title"
      }
    },
    "metadataExtractorPath": ""
  },
  "conversion": {
    "hardbreak": false,
    "dataview": false,
    "censorText": [],
    "tags": {
      "inline": false,
      "exclude": [],
      "fields": []
    },
    "links": {
      "internal": true,
      "unshared": false,
      "wiki": false,
      "slugify": false
    }
  },
  "embed": {
    "attachments": true,
    "forcePushAttachments": [],
    "keySendFile": [
      "banner"
    ],
    "notes": false,
    "folder": "static/attachments",
    "convertEmbedToLinks": "keep",
    "charConvert": "->",
    "useObsidianFolder": false
  }
}
```

**Comment by Lisandra-Archive** [comment]
Okay, the error doesn't seems to be caused by the image itself. It doesn't seems to find the workflows. Are you sure you use the good name? Maybe try with `yaml` instead of `yml` 

(It's been a while I didn't use this option so…)

For the image, it's maybe my code, as I have the tendency to use only the name for the image, and not the entire path. I will check with updating/testing. 

Oh, could you please try by force push the image with editing this: 
![image](https://github.com/ObsidianPublisher/obsidian-github-publisher/assets/118466140/f4b4ed19-17f5-4dd9-bb6b-4f575d7d8d34) and adding "png" ?

In fact, the scripts works as follow for image:
- Is the image exists in the repo ? → Check the date → If the edit and push time are different, push the new attachments.

**Comment by Artlesbol** [comment]
I tried, but it didn't work.
I checked my workflow file name. It is hugo.yaml. but the program seems try to find hugo.yml. then it gets a 404 return code.
Althrough it report an error message, workflows is triggered successfully. I think this error has no connection with image uploading.
![image](https://github.com/ObsidianPublisher/obsidian-github-publisher/assets/74887228/fa5a2ad3-d441-4289-8ece-a046b6749217)
I tried to force push the image while editing this, but the commit says it has no effect.
![image](https://github.com/ObsidianPublisher/obsidian-github-publisher/assets/74887228/a7adac33-d39a-4f2e-8f50-16c9e614e732)
![image](https://github.com/ObsidianPublisher/obsidian-github-publisher/assets/74887228/97410b05-a0d7-4f60-8d5f-5ccd85558cde)

I am sure the image is not exists in the repo

**Comment by Mara-Li** [comment]
Hmm, is your image a png or jpeg/jpg ?

**Reply by Artlesbol** [reply]
yes. it's png

**Comment by Artlesbol** [comment]
I tried to add some logs to your script. I found that the script got files at first but finally returned an empty list. I am not good at javascript or typescript, but I hope this information can help you.
![image](https://github.com/ObsidianPublisher/obsidian-github-publisher/assets/74887228/2326fe83-9844-4c25-b151-055ee5af8ea4)
![image](https://github.com/ObsidianPublisher/obsidian-github-publisher/assets/74887228/c36cfbb3-75a6-4def-821d-1d058323d9c0)
![image](https://github.com/ObsidianPublisher/obsidian-github-publisher/assets/74887228/9c07c3f2-94fb-4c1e-a9f2-a2045e7ba7aa)

**Comment by Mara-Li** [comment]
It's pretty strange because I can't reproduce the issue.

**Reply by Artlesbol** [reply]
Could you show me your obsidian file structure and frontmatter? I want to try it in my local environment again.

**Reply by Mara-Li** [reply]
My only theory is that Obsidian struggle with `getFirstLinkDestinationPath` when the file is not in latin text

**Reply by Mara-Li** [reply]
I pushed a new version that rely on other things to get the file, but I don't think it will change something.

**Reply by Artlesbol** [reply]
Well, as you expect, the new version changes nothing.
I give up solving this strange problem and upload the banner image manually.
But I still thank you for your patience. Merry Christmas❤️

**Reply by Mara-Li** [reply]
I'm sorry that I didn't found the cause of the error ;(

**Reply by Mara-Li** [reply]
Okay, I maybe find the solution to your problem.
There was 2 causes : 
- A typo that I didn't see, for a strange reason tbh 
- If the file already exists in any commit (including deleted after), the repository considered it like existing and skip the upload.
